### ISSUE TITLE
Pass Server and Client to server_* hooks

### ISSUE THREAD

**Issue by Prinzhorn** [mitmproxy/mitmproxy#4546]
#### Problem Description

The `server_*` events are the only ones that get this special treatment of receiving `*HookData`. I thought something was broken at first.

![Selection_851](https://user-images.githubusercontent.com/679144/113683708-c8b51880-96c4-11eb-91b1-0fd65308df8e.png)

#### Proposal

Pass two arguments `Server` and `Client` instead of wrapping them.

#### Additional context

I like symmetrical things. I was expecting `Server` for `server_*` the same way I get `Client` for `client_*`.

**Comment by Prinzhorn** [comment]
> the only ones that get this special treatment of receiving `*HookData`

Or maybe not? It's just differently named, should be `ClientHelloHookData` and `TlsStartHookData`?

![Selection_852](https://user-images.githubusercontent.com/679144/113685109-31e95b80-96c6-11eb-8ede-0ebb33850fb0.png)

They are also not linked properly. I'm trying to migrate to v7 right now, so don't mind me dumping all the things I find along the way :sweat_smile:

**Comment by mhils** [comment]
Yeah - we currently only support passing one argument to hooks, so there is the occasional "data" wrapper. This is a bit unergonomic for `server_*`. For `tls_clienthello` it's a bit more involved, `ClientHelloData.establish_server_tls_first` is just a bool and needs to be wrapped _somehow_ so that it can be modified.

**Comment by Prinzhorn** [comment]
> we currently only support passing one argument to hooks

Shame :smile: 

> For `tls_clienthello` it's a bit more involved, `ClientHelloData.establish_server_tls_first` is just a bool and needs to be wrapped _somehow_ so that it can be modified.

I will get to those too. Not quite sure they do what [I need](#3884).
